test-app/tests/integration/components/headless-form-test.gts

```diff
@@ -412,4 +446,51 @@ module('Integration Component headless-form', function (hooks) {
       });
     });
   });
+
+  module('Glint', function () {
+    // These tests are not testing any new run-time behaviour that isn't tested elsewhere already.
+    // Rather they are here to make sure they pass glint checks, testing for their types constraints to work as expected
+    // Note: @glint-expect-error behaves just as @ts-expect-error in that in surpresses an error when we *expect* it to error,
+    // but it *also* fails when no expected error is actually present!
+
+    test('@name argument only expects keys of @data', async function (assert) {
+      assert.expect(0);
+      const data = { firstName: 'Simon' };
```

Comment:
Suggestion, nit: I'm a bit slow this morning, but it took me a few reads to see that this errors because `lastName` is not supplied to `data`.

Maybe leave some comments here along the lines of `note that `lastName` does not exist here?`.

Reply:
Added :+1: